File: trainease/src/main/java/com/trainease/service/BatchService.java
package com.trainease.service;

import com.trainease.entity.Batch;

import java.util.List;

public interface BatchService {
    List<Batch> getAllBatches();

    Batch getBatchByBatchId(String batchId);

    Batch createBatch(Batch batch);

    Batch updateBatch(Batch batch);

}

File: trainease/web-app/src/app/services/batch.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Batch } from '../models/batch.model';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class BatchService {
  baseUrl = environment.baseUrl;
  isProduction = environment.production;

  constructor(private http: HttpClient) { }

  getAllBatches(): Observable<Batch[]> {
    return this.http.get<Batch[]>(`${this.baseUrl}/batches`);
  }

  createBatch(batch: Batch): Observable<Batch> {
    return this.http.post<Batch>(`${this.baseUrl}/batches`, batch);
  }

  updateBatch(batch: Batch): Observable<Batch> {
    return this.http.put<Batch>(`${this.baseUrl}/batches`, batch);
  }

}
